tooling/debugger/src/context.rs

```diff
@@ -68,25 +85,35 @@ impl AddressMap {
     ) -> Self {
         let opcode_address_size = |opcode: &Opcode<FieldElement>| {
             if let Opcode::BrilligCall { id, .. } = opcode {
-                unconstrained_functions[id.as_usize()].bytecode.len()
+                (unconstrained_functions[id.as_usize()].bytecode.len(), Some(*id))
             } else {
-                1
+                (1, None)
             }
         };
 
         let mut addresses = Vec::with_capacity(circuits.len());
         let mut next_address = 0usize;
+        let mut brillig_addresses = Vec::new();
 
         for circuit in circuits {
             let mut circuit_addresses = Vec::with_capacity(circuit.opcodes.len());
             for opcode in &circuit.opcodes {
                 circuit_addresses.push(next_address);
-                next_address += opcode_address_size(opcode);
+                let (address_size, brillig_function_id) = opcode_address_size(opcode);
+                if let Some(brillig_function_id) = brillig_function_id {
+                    let brillig_address_space = BrilligAddressSpace {
+                        start_address: next_address,
+                        end_address: next_address + address_size,
+                        brillig_function_id,
+                    };
+                    brillig_addresses.push(brillig_address_space);
+                }
+                next_address += address_size;
```

Comment:
> IMHO `opcode_address_size` function does not feel cohesive. It feels like the only reason why the id is there is to avoid repeating the match clause

I agree :)! I was looking at most of the comments and they do look mostly like style suggestions. Would we be good with merging to unblock #5703 and then you could submit a follow-up to improve anything regarding style?